tools/build.py

```diff
@@ -0,0 +1,681 @@
+#!/usr/bin/env python3
```

Comment:
would be good if we can prune this further to what we want to support in ORT extensions. ideally we can inherit only as much complexity as we need from ORT's build.py

Reply:
Pruned more stuff.